core/src/main/java/org/opensearch/sql/storage/StorageEngine.java

```diff
@@ -30,4 +32,8 @@ default Collection<FunctionResolver> getFunctions() {
     return Collections.emptyList();
   }
 
+  default TableScanOperator getTableScan(String indexName, String scrollId) {
```

Comment:
https://github.com/Bit-Quill/opensearch-project-sql/pull/226#discussion_r1108826262 by @penghuo:
> why get TableScanOperator in StorageEngine?

@Yury-Fridlyand:
> It is used in cursor deserialization/parsing, this happens in :core module. All implementation of StorageEngine are defined outside of :core and unavailable and that point.

@MaxKsyunz:
> @penghuo does this design make sense? Would love to know what you think.

Reply:
This is reverted in #1498